server/routes/prisonerSearch.ts

```diff
@@ -122,6 +124,19 @@ export default function prisonerSearchRoutes(service: Services): Router {
             response.totalElements -= 1
             return false
           })
+
+          const api = new NonAssociationsApi(systemToken)
+          const openNonAssociations = await api.listNonAssociationsBetween([
+            prisonerNumber,
+            ...response.content.map(somePrisoner => somePrisoner.prisonerNumber),
+          ])
+          openNonAssociations?.forEach(nonAssociation => {
+            if (nonAssociation.firstPrisonerNumber === prisonerNumber) {
+              openNonAssociationsMap.set(nonAssociation.secondPrisonerNumber, nonAssociation.id)
+            } else if (nonAssociation.secondPrisonerNumber === prisonerNumber) {
+              openNonAssociationsMap.set(nonAssociation.firstPrisonerNumber, nonAssociation.id)
```

Comment:
What about something like?

```JavaScript
let otherPrisonerNumber = [nonAssociation.firstPrisonerNumber, nonAssociation.secondPrisonerNumber].find(pn => pn !== prisonerNumber)
openNonAssociationsMap.set(otherPrisonerNumber, nonAssociation.id)
```

Maybe too clever. Or maybe use ternary operator 🤔 

```JavaScript
let otherPrisonerNumber = nonAssociation.firstPrisonerNumber === prisonerNumber ? nonAssociation.secondPrisonerNumber : nonAssociation.firstPrisonerNumber
openNonAssociationsMap.set(otherPrisonerNumber, nonAssociation.id)
```

Doesn't really matter, just a couple of ideas.

Reply:
only one `if` branch so not tooo bad. my first version added both and then deleted the key prisoner at the end. i'm happy with your second version…

Reply:
oopsy. i forgot that the list of non-associations here might not include the key prisoner at all. thankfully a test caught that 💪

Reply:
Sorry, my bad. This is not an `if`/`else` so the ternary operator doesn't make sense in this case.
Also the non-associations returned from the API (between endpoint) may be between 2 other prisoners, e.g. not involving the `prisonerNumber` one, so what I suggested didn't make sense in this context, soz :)